src/main/java/org/kizombadev/markdownparser/SyntaxAnalyzer.java

```diff
@@ -1,25 +1,32 @@
 package org.kizombadev.markdownparser;
 
 import com.google.common.collect.ImmutableList;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 import org.kizombadev.markdownparser.entities.*;
+import org.kizombadev.markdownparser.entities.interfaces.ImmutableSyntax;
+import org.kizombadev.markdownparser.entities.interfaces.MutableSyntax;
 
 import java.util.ArrayDeque;
 import java.util.Deque;
 
+//todo make the sytaxTree Immutable
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)